### ISSUE TITLE
Failure of BERT model instructions

### ISSUE THREAD

**Issue by fbbradheintz** [pytorch/serve#875]
Instructions for the BERT example at https://github.com/shrinath-suresh/serve/tree/bert-news-classification-example/kubernetes/kfserving/bert_news_classification fails with an error due to mismatched input sizes:

```
RuntimeError: stack expects each tensor to be equal size, but got [27] at entry 0 and [33] at entry 1
```

cc @shrinath-suresh 

## Context
* torchserve version: latest bert-news-classification-example branch at https://github.com/shrinath-suresh/serve/tree/bert-news-classification-example/kubernetes/kfserving/bert_news_classification
* torch version: 1.6.0
* torchtext version [if any]: 0.7.0
* java version: 11
* Operating System and version: macOS 10.15.7

## Your Environment

* Installed using source? [yes/no]: YES
* Are you planning to deploy it using docker container? [yes/no]: YES
* Is it a CPU or GPU environment?: CPU
* Using a default/custom handler? [If possible upload/share custom handler/model]: n/a - didn't get past training in example
* What kind of model is it e.g. vision, text, audio?: BERT text
* Are you planning to use local models from model-store or public url being used e.g. from S3 bucket etc.?
  [If public url then provide link.]: n/a
* Provide config.properties, logs [ts.log] and parameters used for model registration/update APIs: n/a
* Link to your project [if any]: n/a

## Expected Behavior
The failing step was for training the model; I'm assuming that if it behaved correctly, I'd see training status updates (batches, epochs, &c)

## Current Behavior
Failure on mismatched input sizes - see log below

## Possible Solution
???

## Steps to Reproduce
1. `git clone https://github.com/shrinath-suresh/serve`
2. `git checkout bert-news-classification-example`
3. Install TorchServe from source using instructions from main README
4. `cd serve/kubernetes/kfserving/bert_news_classification`
5. `python news_classifier.py --max_epochs 5 --num_samples 50000`

## Failure Logs [if any]
```
(tsahod8) --- kfserving/bert_news_classification ‹bert-news-classification-example› » python news_classifier.py --max_epochs 5 --num_samples 50000
ag_news_csv.tar.gz: 11.8MB [00:02, 4.09MB/s]
Downloading: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 440M/440M [01:07<00:00, 6.49MB/s]
/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py:49: UserWarning: Checkpoint directory /Users/bradheintz/workspace/kfstest/bert_test/serve/kubernetes/kfserving/bert_news_classification exists and is not empty. With save_top_k=1, all files in this directory will be deleted when a checkpoint is saved!
  warnings.warn(*args, **kwargs)
GPU available: False, used: False
INFO:lightning:GPU available: False, used: False
TPU available: None, using: 0 TPU cores
INFO:lightning:TPU available: None, using: 0 TPU cores

  | Name       | Type      | Params
-----------------------------------------
0 | train_acc  | Accuracy  | 0
1 | val_acc    | Accuracy  | 0
2 | test_acc   | Accuracy  | 0
3 | bert_model | BertModel | 109 M
4 | drop       | Dropout   | 0
5 | fc1        | Linear    | 393 K
6 | out        | Linear    | 2.1 K
-----------------------------------------
395 K     Trainable params
109 M     Non-trainable params
109 M     Total params
INFO:lightning:
  | Name       | Type      | Params
-----------------------------------------
0 | train_acc  | Accuracy  | 0
1 | val_acc    | Accuracy  | 0
2 | test_acc   | Accuracy  | 0
3 | bert_model | BertModel | 109 M
4 | drop       | Dropout   | 0
5 | fc1        | Linear    | 393 K
6 | out        | Linear    | 2.1 K
-----------------------------------------
395 K     Trainable params
109 M     Non-trainable params
109 M     Total params
Validation sanity check: 0it [00:00, ?it/s]Traceback (most recent call last):
  File "news_classifier.py", line 414, in <module>
    trainer.fit(model, dm)
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 470, in fit
    results = self.accelerator_backend.train()
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/pytorch_lightning/accelerators/cpu_accelerator.py", line 61, in train
    results = self.train_or_test()
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/pytorch_lightning/accelerators/accelerator.py", line 65, in train_or_test
    results = self.trainer.train()
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 492, in train
    self.run_sanity_check(self.get_model())
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 690, in run_sanity_check
    _, eval_results = self.run_evaluation(test_mode=False, max_batches=self.num_sanity_val_batches)
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 593, in run_evaluation
    for batch_idx, batch in enumerate(dataloader):
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 363, in __next__
    data = self._next_data()
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 989, in _next_data
    return self._process_data(data)
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 1014, in _process_data
    data.reraise()
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/_utils.py", line 395, in reraise
    raise self.exc_type(msg)
RuntimeError: Caught RuntimeError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/_utils/worker.py", line 185, in _worker_loop
    data = fetcher.fetch(index)
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 47, in fetch
    return self.collate_fn(data)
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/_utils/collate.py", line 74, in default_collate
    return {key: default_collate([d[key] for d in batch]) for key in elem}
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/_utils/collate.py", line 74, in <dictcomp>
    return {key: default_collate([d[key] for d in batch]) for key in elem}
  File "/Users/bradheintz/anaconda2/envs/tsahod8/lib/python3.8/site-packages/torch/utils/data/_utils/collate.py", line 55, in default_collate
    return torch.stack(batch, 0, out=out)
RuntimeError: stack expects each tensor to be equal size, but got [27] at entry 0 and [33] at entry 1
```

**Comment by dhanainme** [comment]
@fbbradheintz , I assume this uses the transformers lib. 
Can you please share the transformer lib version ?

Asking coz we have run in a lot of issues with transformer lib version

**Comment by fbbradheintz** [comment]
@dhanainme If you look at the logs, you can see it fails in the DataLoader - I don't believe the transformer comes into it.

Has anyone actually witnessed these instructions work? If so, what version of PyTorch was used? That might make a difference to DataLoader behavior.

**Comment by shrinath-suresh** [comment]
I am able to reproduce the error after compiling from source. Looks like the transformer version is causing the error. I will investigate on it. 

To unblock, the script is working with `transformers==3.0.2` . I upgraded all other packages to latest version (`torch`, `torchtext` and `pytorch_lightning`)..

Please find the environment info captured using `print_env_info.py`

```
------------------------------------------------------------------------------------------
Environment headers
------------------------------------------------------------------------------------------
Torchserve branch: bert-news-classification-example

torchserve==0.2.0
torch-model-archiver==0.2.0

Python version: 3.8 (64-bit runtime)
Python executable: /home/ubuntu/anaconda3/envs/pytorch_p38_torch1_7/bin/python

Versions of relevant python libraries:
numpy==1.19.2
torch==1.7.1
torchtext==0.8.1
torchvision==0.8.2
torchaudio==0.7.0a0+a853dff

Java Version:
openjdk 11.0.9.1 2020-11-04
OpenJDK Runtime Environment (build 11.0.9.1+1-Ubuntu-0ubuntu1.18.04)
OpenJDK 64-Bit Server VM (build 11.0.9.1+1-Ubuntu-0ubuntu1.18.04, mixed mode, sharing)

OS: Ubuntu 18.04.4 LTS
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: N/A
CMake version: version 3.13.3
```

Packages which are not captured by the script:
```
transformers==3.0.2
pytorch_lightning==1.1.0
```

**Comment by shrinath-suresh** [comment]
@fbbradheintz  The issue is due to `transformers` library. I have updated the script to work with the latest version of `transformers` as well. 

I am able to train and test using `3.5.1` and `4.0.1`. Please check and let me know if you face any more issues on the same.

**Comment by fbbradheintz** [comment]
@shrinath-suresh Thanks for the update - I'll retry when I have a chance.

**Comment by HamidShojanazeri** [comment]
@shrinath-suresh @fbbradheintz seems the issue has been resolved with support for transformers>3.0, I could successfully test it with transformers==3.0.2 and 4.3.2. 
[issue_875.logs.txt](https://github.com/pytorch/serve/files/5992724/issue_875.logs.txt)
